src/main/java/com/example/demo1/controller/ProductsController.java

```diff
@@ -16,48 +17,42 @@ public class ProductsController {
 
     @Autowired
     private final ProductService productService;
+    @Autowired
+    private final ProductsRepository productsRepository;
 
-    @GetMapping(value = "/products")
+    @GetMapping(value = "/productList")
     public String listAllProducts(Model model) {
-        model.addAttribute("products", productService.getAllProducts());
-        return "productsPage";
+        model.addAttribute("products", productsRepository.findAll());
+        return "products/productListPage";
     }
 
     @GetMapping(value = "/product/productPage")
     public String productPage(@RequestParam(name = "id") int productId, Model model) {
         if(productService.productExists(productId)) {
-            model.addAttribute("product", productService.getProductById(productId));
+            model.addAttribute("product", productsRepository.findById(productId));
         }
-        return "productPage";
+        return "products/productPage";
     }
 
-    @GetMapping(value = "/product/{id}")
+    @GetMapping(value = "/product/updateProduct/{id}")
     public String edit_product_form(@PathVariable(name = "id") int productId, Model model) {
-        model.addAttribute("product", productService.getProductById(productId));
-        return "updateProductPage";
+        if(productService.productExists(productId)) {
+            model.addAttribute("product", productsRepository.findById(productId));
+            return "products/updateProductPage";
+        } else {
+            return "redirect:/productList";
+        }
     }
 
     @PostMapping(value = "/product/update")
     public String updateProduct(@RequestParam(name = "id") int productId, @ModelAttribute("product") Product newProduct,  RedirectAttributes redirectAttributes) {
-
         if(productService.productExists(productId)) {
-            Product oldProduct = productService.getProductById(productId);
-
-//            oldProduct.setId(newProduct.getId());
-//            oldProduct.setName(newProduct.getName());
-//            oldProduct.setPrice(newProduct.getPrice());
-//            oldProduct.setStock(newProduct.getStock());
-//            oldProduct.setDescription(newProduct.getDescription());
-
-            productService.updateProduct(oldProduct, newProduct);
+            productsRepository.save(newProduct);
```

Comment:
add validation and a service where you work with the database and do processing. don't call the db in the controller. Also use DTOs to get and return data